Svc/BufferManager/BufferManagerComponentImpl.cpp

```diff
@@ -82,9 +82,11 @@ namespace Svc {
       // make sure component has been set up
       FW_ASSERT(this->m_setup);
       FW_ASSERT(m_buffers);
-      // check for empty buffers - this is just a warning since this component returns
-      // empty buffers if it can't allocate one.
-      if (fwBuffer.getSize() == 0) {
+      // check for null, empty buffers - this is a warning because this component returns
+      // null, empty buffers if it can't allocate one.
+      // however, empty non-null buffers could potentially be previously allocated
+      // buffers with their size reduced. the user is allowed to make buffers smaller.
+      if (fwBuffer.getData() == nullptr && fwBuffer.getSize() == 0) {
```

Comment:
Do we need the size == 0 check here? It seems that if the pointer is null, there is nothing to deallocate, regardless of the size.

Reply:
I include both checks for the sake of defensiveness. If the pointer is null but the size is non-zero, that's likely a serious error and should result in an assertion trip. By excluding the null/non-zero case from this conditional, I make sure it will fall through into the subsequent assertions and cause a FATAL. So I don't think I should change this.

Reply:
Thanks, that makes sense!